midend/local_copyprop.h

```diff
@@ -23,15 +23,21 @@ limitations under the License.
 
 namespace P4 {
 
-/* Local copy propagation and dead code elimination within a single action function.
-   This pass is designed to be run after all declarations have received unique
-   internal names.  This is important because the locals map uses only the
-   declaration name, and not the full path.
-
-   Requires expression types be stored inline in the expression
-   (obtained by running Typechecking(updateProgram = true)).
-   Requires that all declaration names be globally unique
-   (obtained by running UniqueNames)
+/**
+Local copy propagation and dead code elimination within a single pass.
+This pass is designed to be run after all declarations have received unique
+internal names.  This is important because the locals map uses only the
+declaration name, and not the full path.
+
+@pre
+Requires expression types be stored inline in the expression
+(obtained by running Typechecking(updateProgram = true)).
+
+Requires that all declaration names be globally unique
+(obtained by running UniqueNames).
+
+Requires that all variable declarations are at the top-level control scope
+(obtained using MoveDeclarations).
```

Comment:
This is actually false -- in particular it can only eliminate declarations from actions if they are in the actions.  So it should work correctly either with or without MoveDeclarations, but the things that it can eliminate differ, and generally it will do a better job of eliminating dead code if called before MoveDeclarations.